pkg/image/vulnerability_scan.go

```diff
@@ -0,0 +1,118 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+package image
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"sort"
+	"strconv"
+	"strings"
+
+	buildapi "github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+)
+
+const (
+	VulnerabilityCountLimit = 50 // Number of vulnerabilities to be added to buildrun output.
+)
+
+func RunVulnerabilityScan(ctx context.Context, imagePath string, settings buildapi.VulnerabilityScanOptions) ([]buildapi.Vulnerability, error) {
+
+	trivyArgs := []string{"image", "--quiet", "--format", "json", "--input", imagePath}
+
+	if settings.IgnoreOptions != nil {
+		if settings.IgnoreOptions.Severity != "" {
+			severity := getSeverityStringForTrivyScan(settings.IgnoreOptions.Severity)
+			trivyArgs = append(trivyArgs, "--severity", severity)
+		}
+		if len(settings.IgnoreOptions.Issues) > 0 {
+			// Create a file with vulnerabilities to be ignored.
+			ignoreFile, err := os.CreateTemp("", "ignore")
+			if err != nil {
+				log.Printf("Could not create file for ignored vulnerabilities: %v\n", err)
+				return nil, err
+			}
+			defer os.Remove(ignoreFile.Name())
+			for _, vul := range settings.IgnoreOptions.Issues {
+				_, err := ignoreFile.WriteString(vul + "\n")
+				if err != nil {
+					log.Printf("Error writing to ignore file for vulnerabilities: %v\n", err)
+					return nil, err
```

Comment:
Same as other comment, question about wrapping vs logging.
```suggestion
					return nil, fmt.Errorf("error writing to ignore file for vulnerabilities: %w", err)
```